src/ai/defaultai_warfare.cc

```diff
@@ -714,19 +714,33 @@ bool DefaultAI::check_trainingsites(const Time& gametime) {
 
 	const Widelands::DescriptionIndex enhancement = ts->descr().enhancement();
 
-	if (enhancement != Widelands::INVALID_INDEX && ts_without_trainers_ == 0 && mines_.size() > 3 &&
+	if (enhancement != Widelands::INVALID_INDEX && ts_without_trainers_ == 0 &&
```

Comment:
removed the check for 3 mines as this was only a mediocre indication for an economy that can sustain enhanced trainingsites and made the enhancement solely dependent on building materials (including gold or other valuable Buildmaterials)